Refactor/epw

ymzhang0 opened the pull request:
Now the previous restart logic is removed. Use multiple get_builder_restart_from_XX instead.

Update the EPW parser for more outputs.

EpwAnisoWorkChain now works in intermediate representation (IR) mode, which allows calculation at really low temperature and the results are better. 

I set full bandwidth as default eigenspace for more accurate aniso Tc.

ymzhang0 committed:
Instead of using restart input namespace. I write multiple get_builder_restart_from_XX for the reuse of existing sub-workchains

This is mainly for the adaption of my current phonon workchains.

After the framework of the workchains is more optimized, one should stick to always use one workchain for the high-throughput calculation.

So the useful builder construction is: get_builder_from_protocol and get_builder_restart

ymzhang0 committed:
Add feat: support intermediate representation in anisotropic calculation
Add feat: add parsers for Eliashberg calculations: dos, phdos, phdos_proj, a2f_proj, lambda_FS, lambda_k_pairs

Modify exposed ports. Always expose the last EpwBaseWorkChain in EpwBaseIntpWorkChain at the main outputs port.

Fix bug: EpwSuperConWorkChain, iso, aniso subworkchain should read qfpoints_distance and kfpoints_factor from (converged) a2f subworkchain

Fix bug: builder from member function get_builder_restart will populate all the namespaces, it is explicitly popped now.

Feat: Restart logic of EpwSuperConWorkChain works now

ymzhang0 committed:
Merge branch 'mbercx:main' into refactor/epw

ymzhang0 committed:
Delete irobjs files. It is contained in EPW source code

ymzhang0 committed:
Merge branch 'refactor/epw' of github.com:ymzhang0/aiida-supercon into refactor/epw

ymzhang0 committed:
Change default value for degaussw from 0.01 to 0.025 (default value of EPW)

Fix bug: fix bug of temps in iso and aniso calculations
Fix bug: fix get_builder_restart function in EpwSuperConWorkChain
Fix bug: IR is not compatible with eps_acoustic, I simply pop it from parameter dict.
Fix bug: I use ArrayData to store gap functions with temperature as their keys.
         Since the key of array can't contain '.' I use '_' instead.
         For example:
              '5.0' -> np.ndarray([...]) ---> '5_0' -> np.ndarray([...])

ymzhang0 committed:
before merge

ymzhang0 committed:
before merge

ymzhang0 committed:
fix bugs

ymzhang0 committed:
update doc

ymzhang0 committed:
fix doc

ymzhang0 committed:
update doc

ymzhang0 committed:
update doc

ymzhang0 committed:
update doc

ymzhang0 committed:
update doc

ymzhang0 committed:
update doc, fix bug

mbercx commented on SUBMISSION_CONTROLLER_GUIDE.md:
```diff
@@ -0,0 +1,303 @@
+
+## 概述
+
+AiiDA Submission Controller 是一个用于管理大量AiiDA工作流提交的工具库。它提供了智能的批量提交、并发控制和状态监控功能。
```
Do we want to write the documentation in Chinese? I know there are tools to have both English and Chinese, which would be helpful, but...

ymzhang0 replied:
Werid. I never remember that I have written this. Looks like something generated by AI. My AI will never be good to write everything especially code in English, as I tell it.

I just delete this file.

mbercx commented on aiida_supercon/__pycache__/deprecated/electronphonon.py:
We should add `__pycache__` files to the `.gitignore`, they shouldn't be committed.

mbercx commented on aiida_supercon/tools/calculators.py:
```diff
@@ -29,3 +29,7 @@ def calculate_lambda_omega(frequency: ArrayLike, spectrum: ArrayLike) -> tuple:
 
     return lambda_, omega_log
+
+# @calcfunction
```
Why just add some commented code?

ymzhang0 replied:
It's moved back to superconworkchain now

mbercx commented on aiida_supercon/workflows/a2f.py:
```diff
@@ -4,143 +4,152 @@
-        # spec.output('Tc_allen_dynes', valid_type=orm.Float,
-        #             help='The Allen-Dynes Tc interpolated from the a2f file.')
-
+        # spec.outline(
```
Again, why is this code commented?

ymzhang0 replied:
Deleted

ymzhang0 replied:
By the way I don't expose Tc_allen_dynes now because we can get it by parsing the stdout. I don't quite understand why we want to calculate it again using spectral function given that we have it in the stdout.

mbercx commented on aiida_supercon/workflows/aniso.py:
```diff
@@ -57,48 +83,62 @@ class EpwAnisoWorkChain(EpwBaseIntpWorkChain):
-        )
-        spec.output('a2f', valid_type=orm.XyData,
-            help='The contents of the `.a2f` file.')
+        # spec.outline(
```
Same question.

ymzhang0 replied:
Deleted

mbercx commented on aiida_supercon/workflows/controllers/config.py:
```diff
@@ -0,0 +1,229 @@
+
+# Predefined controller configurations
+CONTROLLER_CONFIGS = {
+    "lumi_debug": {
```
Do we want to add these specific controllers to a general repository?

ymzhang0 replied:
No I'll delete those machine dependent things.

Actually I haven't tested the controllers. I think it's generated by AI again...

mbercx commented on docs/_build/doctrees/api_reference/aiida_supercon.data.doctree:
You shouldn't add the `_build` of the documentation to you commit, that will really blow up the repository. Use the `.gitignore` here.

mbercx reviewed (changes requested):
Thank @ymzhang0! I'm afraid this PR is obviously too big to review properly. I made some quick comments here and there. I understand you want to work quickly, but I would be careful with adding too many things to a repo. If you care about making it maintainable and useable by other people, that is. At least remove the build of the documentation, there is no reason to have that in the repository.

ymzhang0 commented:
BTW I didn't write clearly in the request. But it's not a well-developed code currently. How about we create a refactor/epw branch in this repo instead of just merge it directly into the main branch?

ymzhang0 committed:
block _build, cache

ymzhang0 committed:
Stop tracking __pycache__, _build directories

ymzhang0 committed:
block __pycache__

ymzhang0 committed:
clean unused import

ymzhang0 committed:
fix doc

ymzhang0 committed:
fix deps

ymzhang0 committed:
fix deps

ymzhang0 committed:
fix bug

ymzhang0 committed:
update doc

ymzhang0 committed:
I put most of the parameters into the workchain instead of putting them in the default_inputs.
Only parameters that are really optional are still in the default_inputs now.

I outline a transport workflow containing:

1. A B2W workchain
2. A band interpolation
3. A A2f workchain for spectral function for the evaluation of resistivity in conjunction with the Ziman formula
4. A BTE workchain for solving the Boltzmann transport equation.

ymzhang0 committed:
One should set up restart type in EpwBaseWorkChain depending on whether parent_folder_epw is provided and what is inside this folder.

I make EpwBaseIntpWorkChain accept a restart_type input so restarting is managed externally, which is not good.

But currently only restart from prefix.ephmat folder is necessary.

I move EpwCalculation from aiida_quantumespresso to aiida_supercon

I delete the modification of parameters inside EpwSuperConWorkChain since now they are taken over inside the sub-processes.

Fix bug in run_iso and run_aniso, update estimated_(an)iso_tc in should_run_(an)iso functions.

Fix bug in run_aniso() in EpwAnisoWorkChain. eps_acustic -> epw_acoustic

ymzhang0 committed:
Modify and complete plot functions. Now they work fairly well.

Update temparetures settings.

Update degaussq definition from phonon band interpolation

Update retrieved_symlink in EpwCalculation to avoid copying of big files/folders

ymzhang0 committed:
Fix should_run_conv: should return True for the first time

ymzhang0 committed:
Fix bugs for EPW 6.0
add conversion from wannier90 mmn file to epw mmn file
delete code_ir since now we can always use the same epw.x code from the begining

Add stability check at ph_base step.

Refactor: Rename package from aiida-supercon to aiida-epw-workflows
